# AMZ-Driverless
# Copyright (c) 2018 Authors:
#   - Juraj Kabzan <kabzanj@gmail.com>
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in all
# copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
# SOFTWARE.

car:

  inertia:
    m:        8350.0   # Weight of the Vehicle [kg]
    m_driver: 0.0     # Weight of the driver   [kg]
    g:        9.81    # Gravitational acc      [m/s^2]
    I_z:      8158    # Moment of inertia  I_zz 

  kinematics:
    l: 6.280         # Vehicle Length [m] (wheelbase 2.984) 
    b_F: 1.205       # From COG to front axle [m]
    b_R: 2.188       # From COG to rear axle [m]
    w_front: 0.644   # Percentage of weight front
    h_cg: 1.0        # Heights of COG [m] (estimate)
    l_width: 2.6     # Width of the vehicle body

    # l_F = l * (1 - w_front);  # Front cg lever arm [m]
    # l_R = l * w_front;  # Rear  cg lever arm [m]

  # Magic Formula:
  # y=D*sin(C*atan(B*(1-E)*x+E*atan(B*x))) equivalent to rajamani
  # y=D*sin(C*atan(B*x-E*(B*x-atan(B*x))))
  tire:
    tire_coefficient: 1.0
    B: 10.0 
    C: 1.9
    D: -1.0 # flip sign to account for differnt coordinate system in fssim
    E: 0.97
    # default coeffs for dry asphalt, updated online

  aero:
    C_Down: # F_Downforce = C_Downforce*v_x^2; C_Downforce = a*b*c
      a: 0.0
      b: 0.0
      c: 0.0
    C_drag: # F_Drag = C_Drag*v_x^2; C_drag = a*b*c
      a: 0.6
      b: 1.0
      c: 1.0

  drivetrain: 
    Cm1: 1
    Cr0: 0
    # bypassing fssim drivetrain dynamics (directly passing Fx (total) as dc) 
    inertia: 9.375 # wheel_plus_packaging ((1/2)*75*0.5^2)
    r_dyn: 0.5
    nm_wheels: 4
  
  torque_vectoring: # deactivated
    K_FFW: 0.0
    K_p: 0.0 
    shrinkage: 1.0
    K_stability: 0.0


